packages/HttpClientTransport/src/HttpClientTransport.js

```diff
@@ -8,12 +8,19 @@ class HttpClientTransport {
     this.sendingOnly = sendingOnly;
     this.messageHandler = new EventEmitter();
     this.hostUrl = hostUrl + ':' + port + HttpClientTransport.ENQUEUE_URI ;
+    this.currentId = 0;
+  }
+
+  generateId() {
+    return Promise.resolve(this.currentId++); //eslint-disable-line
   }
 
   enqueue(queueName, command) {
     this.post(command, (resCommand) => {
-      this.messageHandler.emit(queueName, resCommand);
+      this.messageHandler.emit('msg', resCommand);
```

Comment:
you don't need the emitter.. just store the callback and call it here. also I'm pretty sure `EventEmitter` won't work in browser